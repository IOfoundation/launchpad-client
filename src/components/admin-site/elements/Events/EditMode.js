import React from 'react';
import {Form} from 'formik';
import {PropTypes} from 'prop-types';
import Grid from '@material-ui/core/Grid';

import FormTextField from '@Shared/FormElements/TextFieldDefault';
import Checkbox from '@Shared/FormElements/Checkbox';
import SelectElement from '@Shared/FormElements/Select';

import {states} from '@StaticData/data';

const EditMode = props => {
  const {
    closeClicked,
    errors,
    handleBlur,
    handleChange,
    touched,
    values,
  } = props;

  return (
    <Form className="profile-form">
      <Grid container={true} spacing={16}>
        <Grid item={true} xs={12} md={6}>
          <FormTextField
            autocomplete="off"
            error={touched.title && Boolean(errors.title)}
            errors={errors}
            handleBlur={handleBlur}
            handleChange={handleChange}
            id="title"
            name="title"
            label="Event Title"
            value={values.title}
          />
          <FormTextField
            autocomplete="off"
            error={touched.website && Boolean(errors.website)}
            errors={errors}
            handleBlur={handleBlur}
            handleChange={handleChange}
            id="website"
            name="website"
            label="Event URL"
            value={values.website}
          />
          <FormTextField
            autocomplete="off"
            error={touched.description && Boolean(errors.description)}
            errors={errors}
            handleBlur={handleBlur}
            handleChange={handleChange}
            id={'description'}
            label={'Event Description'}
            multiline={true}
            value={values.description}
            rowsMax="12"
          />
        </Grid>
        <Grid item={true} xs={12} md={6}>
          <FormTextField
            autocomplete="off"
            error={touched.address && Boolean(errors.address)}
            errors={errors}
            handleBlur={handleBlur}
            handleChange={handleChange}
            id="address"
            name="address"
            label="Address"
            value={values.address}
          />
          <FormTextField
            autocomplete="off"
            error={touched.address2 && Boolean(errors.address2)}
            errors={errors}
            handleBlur={handleBlur}
            handleChange={handleChange}
            id="address2"
            name="address2"
            label="Address (Line 2)"
            value={values.address2}
          />
          <Grid container={true} spacing={16}>
            <Grid item={true} xs={12} sm={6}>
              <FormTextField
                autocomplete="off"
                error={touched.city && Boolean(errors.city)}
                errors={errors}
                handleBlur={handleBlur}
                handleChange={handleChange}
                id="city"
                name="city"
                label="City"
                value={values.city}
              />
            </Grid>
            <Grid item={true} xs={12} sm={3}>
              <SelectElement
                errors={errors}
                handleBlur={handleBlur}
                handleChange={handleChange}
                id={'state'}
                label={'State'}
                name={'state'}
                selectOptions={states}
                value={values.state}
              />
            </Grid>
            <Grid item={true} xs={12} sm={3}>
              <FormTextField
                autocomplete="off"
                error={touched.zip && Boolean(errors.zip)}
                errors={errors}
                handleBlur={handleBlur}
                handleChange={handleChange}
                id="zip"
                name="zip"
                label="Zipcode"
                value={values.zip}
              />
            </Grid>
          </Grid>

          <Grid container={true} spacing={16}>
            <Grid item={true} xs={5}>
              <FormTextField
                errors={errors}
                handleBlur={handleBlur}
                handleChange={handleChange}
                id="startDate"
                label="Start Date"
                name="startDate"
                value={values.startDate}
                type="date"
              />
            </Grid>
            <Grid item={true} xs={3}>
              <FormTextField
                autocomplete="off"
                errors={errors}
                handleBlur={handleBlur}
                handleChange={handleChange}
                id="startTime"
                label="Start Time"
                name="startTime"
                value={values.startTime}
                type="time"
              />
            </Grid>
          </Grid>
          <Grid container={true} spacing={16}>
            <Grid item={true} xs={5}>
              <FormTextField
                errors={errors}
                handleBlur={handleBlur}
                handleChange={handleChange}
                id="endDate"
                label="End Date"
                name="endDate"
                value={values.endDate}
                type="date"
              />
            </Grid>
            <Grid item={true} xs={3}>
              <FormTextField
                autocomplete="off"
                errors={errors}
                handleBlur={handleBlur}
                handleChange={handleChange}
                id="endTime"
                label="End Date"
                name="endTime"
                value={values.endTime}
                type="time"
              />
            </Grid>
          </Grid>
          <Checkbox
            label="All Day Event"
            name="allDay"
            onChange={handleChange}
            value={values.allDay}
          />
          <Grid
            container={true}
            spacing={16}
            justify="flex-end"
            alignContent="center"
            alignItems="center"
          >
            <Grid item={true} className="m-x-8">
              <button
                className="btn btn__outline"
                type="button"
                onClick={closeClicked}
              >
                {'Cancel'}
              </button>
            </Grid>
            <Grid item={true}>
              <button className="btn btn__black p-x-24" type="submit">
                {'Create Event'}
              </button>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Form>
  );
};

EditMode.propTypes = {
  closeClicked: PropTypes.func,
  errors: PropTypes.shape({
    fullName: PropTypes.string,
    contactEmail: PropTypes.string,
    currentPassword: PropTypes.string,
    newPassword: PropTypes.string,
    confirmPassword: PropTypes.string,
  }),
  handleBlur: PropTypes.func,
  handleChange: PropTypes.func,
  handleSubmit: PropTypes.func,
  isSubmitting: PropTypes.bool,
  isValid: PropTypes.bool,
  touched: PropTypes.shape({
    fullName: PropTypes.bool,
    contactEmail: PropTypes.bool,
    currentPassword: PropTypes.bool,
    newPassword: PropTypes.bool,
    confirmPassword: PropTypes.bool,
  }),
  values: PropTypes.shape({
    fullName: PropTypes.string,
    contactEmail: PropTypes.string,
    currentPassword: PropTypes.string,
    newPassword: PropTypes.string,
    confirmPassword: PropTypes.string,
  }),
};

export default EditMode;
