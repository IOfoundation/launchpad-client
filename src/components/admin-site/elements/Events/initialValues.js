import {falsyToString} from '@Utils';

const parseDate = hour => {
  const date = hour.split('T');

  return {
    date: date ? date[0] : '',
    time: date ? date[1].split('.')[0] : '',
  };
};

const initialValues = ({
  body,
  city,
  ending_at,
  external_url,
  is_all_day,
  starting_at,
  state_abbr,
  street_1,
  street_2,
  title,
  zip,
}) => {
  const mapped = {};
  const start = parseDate(starting_at);
  const end = parseDate(ending_at);

  mapped.city = falsyToString(city);
  mapped.title = falsyToString(title);
  mapped.website = falsyToString(external_url);
  mapped.description = falsyToString(body);
  mapped.address = falsyToString(street_1);
  mapped.address2 = falsyToString(street_2);
  mapped.state = falsyToString(state_abbr);
  mapped.zip = falsyToString(zip);
  mapped.startDate = start.date;
  mapped.startTime = start.time;
  mapped.endDate = end.date;
  mapped.endTime = end.time;
  mapped.allDay = Boolean(is_all_day);

  return mapped;
};

export default initialValues;
