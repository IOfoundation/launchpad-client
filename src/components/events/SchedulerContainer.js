import React, {PureComponent} from 'react';
import '@progress/kendo-ui';
import '@progress/kendo-ui/js/kendo.timezones';
import {PropTypes} from 'prop-types';
import {Scheduler} from '@progress/kendo-scheduler-react-wrapper';
import {withStyles} from '@material-ui/core/styles';
import kendo from '@progress/kendo-ui';

import {containerStyles, getDate, mobileDaysMap} from '@Utils';

const styles = theme => ({
  container: {
    padding: 12,
    ...containerStyles(theme),
  },
  paper: {
    backgroundColor: theme.palette.background.paper,
    borderRadius: '5px',
    boxShadow: theme.shadows[5],
    left: '50%',
    maxHeight: '70vh',
    outline: 'none',
    overflow: 'auto',
    padding: '32px 24px',
    position: 'absolute',
    top: '45%',
    transform: 'translate(-50%, -50%)',
    width: theme.spacing.unit * 70,
    [theme.breakpoints.down('xs')]: {
      width: '100%',
      boxSizing: 'border-box',
    },
  },
});

class SchedulerContainer extends PureComponent {
  state = {
    views: ['day', 'week', {type: 'month', selected: true}],
    events: [],
  };

  static getDerivedStateFromProps(nextProps, prevState) {
    if (
      nextProps.events.data &&
      nextProps.events.data.length !== prevState.events.length
    ) {
      const mappedEvents = nextProps.events.data.map(event => {
        const start = getDate(event.starting_at);
        const end = getDate(event.ending_at);
        const mapped = {
          ID: event.id,
          title: event.title,
          start: new Date(start.pacificTime),
          end: new Date(end.pacificTime),
          description: event.body,
          recurrenceId: event.organization_id,
        };

        if (event.is_all_day) {
          mapped.isAllDay = event.is_all_day;
        }

        return mapped;
      });

      return {events: mappedEvents};
    }

    return null;
  }

  componentDidMount() {
    this.props.actions.getAllEventsByMonth(this._date.monthNumeric);
    this._schedulerRef.widgetInstance.element[0].addEventListener(
      'click',
      this._schedulerClicked
    );
  }

  componentDidUpdate(prevProps) {
    const {data: nextEvents, loading: nextLoading} = this.props.events;
    const {data: currentEvents, loading: prevLoading} = prevProps.events;

    if (
      this._thereIsNoEventsAndIsNotLoading(
        nextEvents,
        currentEvents,
        nextLoading,
        prevLoading
      )
    ) {
      this.setLoadingStatusOnScheduler(false);
    }
  }

  componentWillUnmount() {
    this._schedulerRef.widgetInstance.element[0].removeEventListener(
      'click',
      this._schedulerClicked
    );
  }

  _schedulerRef = null;
  _schedulerClicked = event => {
    const className = event.target.className;

    if (className === 'k-event-template') {
      this.openModal(event.target.textContent);
    }
  };
  _thereIsNoEventsAndIsNotLoading = (
    nextEvents,
    currentEvents,
    nextLoading,
    prevLoading
  ) => {
    return (
      nextEvents.length === currentEvents.length &&
      nextEvents.length === 0 &&
      nextLoading !== prevLoading &&
      !nextLoading
    );
  };
  _changeName = true;
  _date = getDate();
  _month = this._date.monthNumeric;

  openModal = text => {
    const modalInformation = this.props.events.data.find(
      _event => _event.title === text
    );

    this.props.handlerModalVisibility(modalInformation);
  };

  updateCalendar = event => {
    const pickedDate = getDate(event.sender._model.selectedDate);
    const monthNumeric = pickedDate.date.toLocaleString('en-US', {
      month: 'numeric',
    });

    if (event.sender._selectedViewName === 'week' && this._changeName) {
      const weekDate = this._schedulerRef.widgetInstance.element[0].querySelectorAll(
        '.k-scheduler-header .k-nav-day'
      );

      weekDate.forEach(date => {
        const parse = date.textContent.split(' ');

        date.textContent = `${mobileDaysMap[parse[0]]} ${parse[1]}`;
      });
      this._changeName = false;
    } else if (event.sender._selectedViewName !== 'week') {
      this._changeName = true;
    }

    if (monthNumeric !== this._month) {
      this.setLoadingStatusOnScheduler(true);
      this._month = monthNumeric;
      this.props.actions.getAllEventsByMonth(monthNumeric);
    }
  };

  handlerDataBinding = event => {
    if (event.action === 'rebind' && event.items.length > 0) {
      this.setLoadingStatusOnScheduler(false);
    } else {
      this.setLoadingStatusOnScheduler(true);
    }
  };

  setLoadingStatusOnScheduler = value => {
    const scheduler = this._schedulerRef.widgetInstance.element.getKendoScheduler();
    kendo.ui.progress(scheduler.element.find('.k-scheduler-content'), value);
  };

  render() {
    const {classes, breakpoint} = this.props;
    const {startTime, views, events} = this.state;

    return (
      <div className={classes.container}>
        <Scheduler
          height={660}
          views={views}
          dataSource={events}
          date={this._date.date}
          startTime={startTime}
          mobile={breakpoint === 'xs'}
          editable={false}
          ref={_ref => {
            this._schedulerRef = _ref;
          }}
          dataBound={event => this.updateCalendar(event)}
          dataBinding={event => this.handlerDataBinding(event)}
        />
      </div>
    );
  }
}

SchedulerContainer.propTypes = {
  actions: PropTypes.shape({
    getAllEventsByMonth: PropTypes.func,
  }),
  breakpoint: PropTypes.string,
  classes: PropTypes.shape({
    container: PropTypes.string,
    paper: PropTypes.string,
  }),
  events: PropTypes.shape({
    data: PropTypes.arrayOf(PropTypes.shape({})),
    errors: PropTypes.shape({}),
    loading: PropTypes.bool,
  }),
  handlerModalVisibility: PropTypes.func,
};

export default withStyles(styles)(SchedulerContainer);
