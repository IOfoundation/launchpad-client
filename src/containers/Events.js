import React, {PureComponent} from 'react';
import MainLayout from '../components/layouts/Main';
import EventsLayout from '../components/events/Events';
import {sizeCheck, viewport} from '../utils';

class Events extends PureComponent {
  state = {
    breakpoint: '',
    homePage: false,
    listener: () => sizeCheck(this.handleWindowSizeChange),
    width: viewport().width,
  };

  componentDidMount() {
    window.addEventListener('resize', this.state.listener);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.state.listener);
  }

  handleWindowSizeChange = breakpoint => {
    this.setState({breakpoint, width: viewport().width});
  };

  render() {
    const {width, homePage, breakpoint} = this.state;

    return (
      <MainLayout windowWidth={width} homePage={homePage}>
        <EventsLayout breakpoint={breakpoint} />
      </MainLayout>
    );
  }
}

export default Events;
