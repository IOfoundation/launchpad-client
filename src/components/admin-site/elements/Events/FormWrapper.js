import React, {Fragment} from 'react';
import {Formik} from 'formik';
import * as Yup from 'yup';
import {PropTypes} from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {withRouter} from 'react-router';

import EditMode from './EditMode';

import * as user from '@Actions/user';
import * as snackbarActions from '@Actions/snackbar';

export const FormSchema = Yup.object().shape({
  title: Yup.string().required('Required'),
  website: Yup.string()
    .url('URL invalid')
    .notRequired(),
  description: Yup.string(),
  address: Yup.string(),
  address2: Yup.string(),
  city: Yup.string(),
  state: Yup.string(),
  zip: Yup.string(),
  startDate: Yup.string(),
  endDate: Yup.string(),
  startTime: Yup.string(),
  endTime: Yup.string(),
  allDay: Yup.string(),
});

const initialValues = {
  title: '',
  website: '',
  description: '',
  address: '',
  address2: '',
  city: '',
  state: '',
  zip: '',
  startDate: '',
  endDate: '',
  startTime: '',
  endTime: '',
  allDay: '',
};

const FormWrapper = props => {
  const {breakpoint, closeClicked} = props;

  return (
    <Formik
      enableReinitialize={true}
      render={_props => (
        <Fragment>
          <h1 className="events-form-title">{'Create An Event'}</h1>
          <EditMode
            {..._props}
            breakpoint={breakpoint}
            closeClicked={closeClicked}
          />
        </Fragment>
      )}
      initialValues={initialValues}
      validationSchema={FormSchema}
      onSubmit={() => {}}
    />
  );
};

const mapStateToProps = _state => {
  const organizationId =
    _state.user.organizationId || sessionStorage.getItem('organizationId');
  const auth = _state.user.authorization || sessionStorage.getItem('userAuth');

  return {
    auth,
    organizationId,
  };
};

const mapDispatchToProps = _dispatch => {
  return {
    userActions: bindActionCreators(user, _dispatch),
    snackbar: bindActionCreators(snackbarActions, _dispatch),
  };
};

FormWrapper.propTypes = {
  auth: PropTypes.string,
  breakpoint: PropTypes.string,
  business: PropTypes.shape({
    fetchOrganizationById: PropTypes.func,
  }),
  closeClicked: PropTypes.func,
  errors: PropTypes.arrayOf(PropTypes.shape({})),
  isAuth: PropTypes.bool,
  organization: PropTypes.shape({}),
  organizationId: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  profile: PropTypes.shape({
    updateCompany: PropTypes.func,
  }),
  router: PropTypes.shape({
    push: PropTypes.func,
  }),
  snackbar: PropTypes.shape({
    showSnackbar: PropTypes.func,
  }),
  success: PropTypes.bool,
  updatedOrganization: PropTypes.shape({}),
  userActions: PropTypes.shape({
    login: PropTypes.func,
  }),
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(FormWrapper));
