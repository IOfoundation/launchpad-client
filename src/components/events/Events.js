import React from 'react';
import {PropTypes} from 'prop-types';

import Jumbotron from './Jumbotron';
import FeaturedEvents from './FeaturedEvents/FeaturedEvents';
import SchedulerContainer from './SchedulerContainer';

const Events = props => {
  const {breakpoint} = props;

  return (
    <div className="events-container">
      <Jumbotron />
      <FeaturedEvents />
      <SchedulerContainer breakpoint={breakpoint} />
    </div>
  );
};

Events.propTypes = {
  breakpoint: PropTypes.string,
};

export default Events;
