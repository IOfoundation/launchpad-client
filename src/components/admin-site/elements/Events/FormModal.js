import React from 'react';
import ModalUi from '@material-ui/core/Modal';
import {withStyles} from '@material-ui/core/styles';
import {PropTypes} from 'prop-types';

import FormWrapper from './FormWrapper';
import Loading from '@Shared/Loading';

const FormModal = props => {
  const {
    openModal,
    handlerModalVisibility,
    classes,
    refreshData,
    selectedEvent,
    dataLoading,
  } = props;
  let form = <Loading />;

  if (!dataLoading) {
    form = (
      <FormWrapper
        closeClicked={handlerModalVisibility}
        refreshData={refreshData}
        selectedEvent={selectedEvent}
      />
    );
  }

  return (
    <ModalUi open={openModal} onClose={handlerModalVisibility}>
      <div className={classes.paper}>
        <i
          className="material-icons modal-events__close"
          onClick={handlerModalVisibility}
        >
          {'close'}
        </i>
        {form}
      </div>
    </ModalUi>
  );
};

const styles = theme => ({
  paper: {
    backgroundColor: theme.palette.background.paper,
    borderRadius: '5px',
    boxShadow: theme.shadows[5],
    left: '50%',
    maxHeight: '70vh',
    outline: 'none',
    overflowX: 'hidden',
    overflowY: 'auto',
    padding: '32px 24px',
    position: 'absolute',
    top: '45%',
    transform: 'translate(-50%, -50%)',
    width: '100%',
    maxWidth: '930px',
    boxSizing: 'border-box',
    [theme.breakpoints.down('sm')]: {
      width: '100%',
    },
  },
});

FormModal.propTypes = {
  classes: PropTypes.shape({
    paper: PropTypes.string,
  }),
  dataLoading: PropTypes.bool,
  handlerModalVisibility: PropTypes.func,
  openModal: PropTypes.bool,
  refreshData: PropTypes.func,
  selectedEvent: PropTypes.shape({}),
};

export default withStyles(styles)(FormModal);
