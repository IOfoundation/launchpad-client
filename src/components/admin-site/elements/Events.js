import React, {PureComponent} from 'react';
import {PropTypes} from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import LandingComponent from '../Landing';
import Title from '../Title';
import Items from './Events/Items';
import Pagination from '../../businesses/Pagination';
import CustomTabs from '@Shared/Tabs';
import Loading from '@Shared/Loading';

import * as eventActions from '@Actions/events';
import {htmlStripper, truncate, getDate} from '@Utils';

class Events extends PureComponent {
  componentDidMount() {
    this.getEvents();
  }

  menuChanged = index => {
    this._tabSelected = this._tabOptions[index];
  };

  handleChangePage = () => {};

  getEvents = () => {
    this.props.actions.getAllEvents();
  };

  _getPagination = (page, totalPages) => {
    return (
      <Pagination
        appliedFilters={{category: 'admin-posts', page}}
        handleChangePage={this.handleChangePage}
        metadata={{
          pagination: {
            last: {
              page: totalPages,
            },
            currentPage: page,
          },
          totalOrganization: String(totalPages),
        }}
        noMargin={true}
      />
    );
  };

  _tabOptions = ['Upcoming', 'Past Events'];
  _tabSelected = 'Upcoming';

  render() {
    const {events, noResults, loading} = this.props;
    let upcomingElements = <Loading />;
    let pastEventsElements = <Loading />;
    const pagination = null;

    if (noResults && !loading) {
      upcomingElements = (
        <p className="text-regular paragraph">{'No posts available.'}</p>
      );
      pastEventsElements = (
        <p className="text-regular paragraph">{'No posts available.'}</p>
      );
    } else if (events.length > 0 && !loading) {
      upcomingElements = <Items items={events} disable={false} />;
      pastEventsElements = <Items items={events} disable={true} />;
    }

    return (
      <LandingComponent navigation={true}>
        <Title
          titleText="Your Events"
          hideCancelAction={true}
          submitLabel="Create an Event"
        />
        <CustomTabs tabs={this._tabOptions} changed={this.menuChanged}>
          {upcomingElements}
          {pastEventsElements}
        </CustomTabs>
        {pagination}
      </LandingComponent>
    );
  }
}

function eventsToItemsProps(events) {
  return events.map(event => {
    const date = getDate(event.posted_at);
    const start = getDate(event.starting_at);
    let description = htmlStripper(event.body);
    let title = event.title;

    if (description.split('').length > 190) {
      description = truncate(description, 190);
    }

    if (title.split('').length > 130) {
      title = truncate(title, 130);
    }

    return {
      id: event.id,
      title,
      description,
      label: 'Posted:',
      date: `${date.monthLarge} ${date.dayNumber}, ${date.year}`,
      start: `${start.monthLarge} ${start.dayNumber}, ${start.year}`,
      category: '',
      url: event.external_url,
    };
  });
}

const mapStateToProps = _state => {
  const {events} = _state.events;
  return {
    events: eventsToItemsProps(events.data),
    noResults: events.data.length === 0,
    loading: events.loading,
  };
};

const mapDispatchToProps = _dispatch => {
  return {
    actions: bindActionCreators(eventActions, _dispatch),
  };
};

Events.propTypes = {
  actions: PropTypes.shape({
    getAllEvents: PropTypes.func,
  }),
  events: PropTypes.arrayOf(PropTypes.shape({})),
  loading: PropTypes.bool,
  noResults: PropTypes.bool,
  router: PropTypes.shape({
    push: PropTypes.func,
  }),
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Events);
