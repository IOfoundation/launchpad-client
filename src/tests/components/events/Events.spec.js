import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Events from '../../../components/events/Events';
import SchedulerContainer from '../../../components/events/SchedulerContainer';

const props = {
  breakpoint: 'sm',
};

describe('<Events />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Events {...props} />);
  });

  it('renders snapshot of Events', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('SchedulerContainer', () => {
    const schedulerContainer = wrapper.find(SchedulerContainer);

    expect(schedulerContainer.props()).toEqual({breakpoint: 'sm'});
  });
});
