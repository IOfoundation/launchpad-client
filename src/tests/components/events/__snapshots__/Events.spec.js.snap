// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Events /> renders snapshot of Events 1`] = `
<div
  className="events-container"
>
  <Jumbotron />
  <WithStyles(Connect(FeaturedEvents)) />
  <WithStyles(Connect(SchedulerContainer))
    breakpoint="sm"
  />
</div>
`;
